custom_components/spotcast/helpers.py

```diff
@@ -103,117 +107,324 @@ async def run(*args, loop=None, executor=None, **kwargs):
 
     return run
 
-def get_search_results(search:str, spotify_client:spotipy.Spotify, country:str=None) -> str:
 
-    _LOGGER.debug("using search query to find uri")
-
-    SEARCH_TYPES = ["artist", "album", "track", "playlist"]
+def get_top_tracks(
+    artistName: str,
+    spotify_client: spotipy.Spotify,
+    limit: int = 20,
+    country: str = None,
+):
 
-    search = search.upper()
+    _LOGGER.debug("Searching for top tracks for the artist: %s", artistName)
+    searchType = "artist"
+    search = searchType + ":" + artistName
 
-    results = []
+    artistUri = ""
 
-    for searchType in SEARCH_TYPES:
-
-        try:
+    # get artist uri
+    try:
 
-            result = spotify_client.search(
-                searchType + ":" + search,
-                limit=1,
-                offset=0,
-                type=searchType,
-                market=country)[searchType + 's']['items'][0]
-
-            results.append(
-                {
-                    'name': result['name'].upper(),
-                    'uri': result['uri']
-                }
+        artist = spotify_client.search(
+            q=search,
+            limit=1,
+            offset=0,
+            type="artist",
+            market=country,
+        )["artists"]["items"][0]
+
+        _LOGGER.debug("found artist %s: %s", artist["name"], artist["uri"])
+        artistUri = artist["uri"]
+
+    except IndexError:
+        pass
+
+    results = spotify_client.artist_top_tracks(artistUri)
+    for track in results["tracks"][:10]:
+        _LOGGER.debug("track    : " + track["name"])
+
+    return results["tracks"]
+
+
+def get_search_string(
+    artistName: str,
+    albumName: str,
+    trackName: str,
+    genreName: str,
+    playlistName: str,
+    showName: str,
+    episodeName: str,
+    audiobookName: str,
+) -> str:
+    search = []
+    if not is_empty_str(artistName):
+        search.append(f"artist:{artistName}")
+        search.append(artistName)
+    if not is_empty_str(albumName):
+        search.append(f"album:{albumName}")
+        search.append(albumName)
```

Comment:
I am not sure yet if this is the right thing to do... Basically, it is a text query, plus a filter, together. Could we have just one?